azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/JoinCallButtonHolderView.kt

```diff
@@ -63,6 +64,14 @@ internal class JoinCallButtonHolderView : ConstraintLayout {
         viewLifecycleOwner.lifecycleScope.launch {
             viewModel.getDisableJoinCallButtonFlow().collect { onDisableJoinCallButtonChanged(it) }
         }
+
+        viewModel.getCallType()?.let {
```

Comment:
ViewModel should rather expose text for the button. Then logic `if (it == CallType.ONE_TO_N_OUTGOING) {...` will be in the ViewModel.

Reply:
I think we should not inject context into view models, as it is view layer responsibility